geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/AlterQueryServiceCommand.java

```diff
@@ -31,78 +29,70 @@
 import org.apache.geode.cache.query.management.configuration.QueryConfigService;
 import org.apache.geode.distributed.ConfigurationPersistenceService;
 import org.apache.geode.distributed.DistributedMember;
-import org.apache.geode.management.cli.CliMetaData;
 import org.apache.geode.management.cli.SingleGfshCommand;
-import org.apache.geode.management.internal.cli.AbstractCliAroundInterceptor;
-import org.apache.geode.management.internal.cli.GfshParseResult;
 import org.apache.geode.management.internal.cli.functions.AlterQueryServiceFunction;
 import org.apache.geode.management.internal.cli.functions.CliFunctionResult;
 import org.apache.geode.management.internal.cli.result.model.ResultModel;
 import org.apache.geode.management.internal.security.ResourceOperation;
 import org.apache.geode.security.ResourcePermission;
 
 public class AlterQueryServiceCommand extends SingleGfshCommand {
-
   static final String COMMAND_NAME = "alter query-service";
-  private static final String COMMAND_HELP =
-      "Alter configuration parameters for the query service";
-  static final String METHOD_AUTHORIZER_NAME = "method-authorizer";
+  private static final String COMMAND_HELP = "Alter configuration parameters for the query service";
+  static final String AUTHORIZER_NAME = "method-authorizer";
   private static final String METHOD_AUTHORIZER_NAME_HELP =
       "The name of the class to be used for OQL method authorization";
   static final String AUTHORIZER_PARAMETERS = "authorizer-parameters";
   private static final String AUTHORIZER_PARAMETERS_HELP =
-      "A semicolon separated list of all parameter values for the specified method authorizer. Requires '--"
-          + METHOD_AUTHORIZER_NAME + "' option to be used";
-  static final String PARAMETERS_WITHOUT_AUTHORIZER_MESSAGE = "The '--" + AUTHORIZER_PARAMETERS
-      + "' option requires '--" + METHOD_AUTHORIZER_NAME + "' to be specified";
-  static final String NO_ARGUMENTS_MESSAGE =
-      "No arguments were provided. No changes have been applied.";
+      "A semicolon separated list of all parameter values for the specified method authorizer.";
+  static final String FORCE_UPDATE = "force-update";
+  private static final String FORCE_PARAMETER_HELP =
+      "Flag to indicate whether to forcibly update the currently configured authorizer, even when there are continuous queries registered (use with caution)";
```

Comment:
I've changed the parameter name, thanks for pointing that out.
I'm not sure about adding the part about invalidating CQs, though, it might be too much for actually being "helpful" to a user (maybe they don't even have CQs at all, adding confusion instead of help). Also, we only invalidate the internal cache used by CQs (internal mechanism totally unknown to our users), I wouldn't know how to phrase this warning without going into much detail and confuse even more the operator...
The user always has the option to go and check the documentation if needed, and the `use with caution` part should be "enough" to make them think twice before using the flag.
Please let me know your thoughts around this.

Reply:
I think you're right that if the cache that's invalidated is entirely internal then we don't need to tell users about it. At that point it's just making things more confusing.